File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 1:
            return 0

        l, r = 0, 1
        res = 0

        while l < r and r < len(prices):
            res = max(res, prices[r] - prices[l])
            while l < r and prices[l] >= prices[r]:
                l += 1
            r += 1
        
        return res

File: Data Structures & Algorithms/maximum-subarray/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxSubArray(nums) {
        let res = nums[0], curSum = 0;
        for (let n of nums) {
            curSum = Math.max(curSum, 0);
            curSum += n;
            res = Math.max(curSum, res);
        }
        return res;
    }
}
